### PR TITLE
Use new generator to generate stubs and publish to PyPI

### PR DESCRIPTION
**Pull request by jmerle**
#### Description

This PR updates the Travis CI configuration so that it uses the new Python stubs generator at [QuantConnect/quantconnect-stubs-generator](https://github.com/QuantConnect/quantconnect-stubs-generator) to generate stubs and so that these stubs are published to PyPI. This is done whenever a new tag is published, and the value of the tag is used as the new version to publish to PyPI.

This PR is marked as a breaking change because:
- I have removed the functionality which uploads the stubs to S3. The new stubs will be installable with `pip install quantconnect-stubs`, which doesn't require Skylight to be installed and works with basically all editors (not just the ones Skylight supports installing the stubs for). Please let me know if you'd like this to stay, I can revert removing the code that is responsible for this (but I cannot test it, because I don't have S3 credentials).
- I have removed the pushing of the stubs to the [Algorithm.Python/stubs](https://github.com/QuantConnect/Lean/tree/master/Algorithm.Python/stubs) directory, because it is no longer needed to maintain this directory if you can also install the stubs with pip. As with the previous item, let me know if you'd like this to stay.

**Important**: Before merging it is important to set the `PYPI_API_TOKEN` environment variable on Travis to a PyPI API token that can be used to publish packages from the QuantConnect account. An API token can be created on [https://pypi.org/manage/account/](https://pypi.org/manage/account/).

#### Related Issue

N/A

#### Motivation and Context

The new stubs generator replaces the IronPython-based generator because the stubs generated by the new generator provide a better developer experience.

#### Requires Documentation Change

It may be useful to add some information about the new stubs along with information on Skylight and the CLI to the [Developing in the IDE > Local Development](https://www.quantconnect.com/docs/key-concepts/developing-in-the-ide#Developing-in-the-IDE-Local-Development) page.

#### How Has This Been Tested?

I have tested this change in [jmerle/Lean/feature-new-stubs-generator-tmp](https://github.com/jmerle/Lean/tree/feature-new-stubs-generator-tmp). The latest Travis build that published a new package to TestPyPI is [#13](https://travis-ci.com/github/jmerle/Lean/builds/196247394).

To test whether the stubs actually work when installing from PyPI I installed them as per the instructions on [https://test.pypi.org/project/quantconnect-stubs/](https://test.pypi.org/project/quantconnect-stubs/) and then opened some example strategies in PyCharm and in VS Code (with the [Pylance](https://marketplace.visualstudio.com/items?itemName=ms-python.vscode-pylance) extension) to verify everthing works as expected.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [ ] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

It's not feasible to automate testing these changes, the new stubs generator got its own tests though.

### CODE REVIEW

**Review by gsalaz98 — CHANGES_REQUESTED** [review]
Just one change and this is good to go 👍 - Very elegant work!

File: ci_build_stubs.sh [hunk 1]
```diff
@@ -1,157 +1,111 @@
...
-    
-    local graphql_query=$(printf '{"query":"mutation { createPullRequest(input: { title: \\"[Travis #%s] Autogenerated Python Stubs\\", baseRefName: \\"%s\\", headRefName: \\"%s\\", maintainerCanModify: true, draft: false, repositoryId: \\"%s\\"}) { pullRequest { resourcePath }}}","variables":{}}' "$TRAVIS_BUILD_NUMBER" "$GH_BASE_BRANCH" "$PUSH_BRANCH" "$GH_REPO_ID")
+    cd $STUBS_DIR
+    python setup.py --quiet sdist bdist_wheel
 
-    curl -s -H "Authorization: bearer $GH_API_KEY" -X POST --data-raw "$graphql_query" https://api.github.com/graphql > /dev/null 2>&1
+    TWINE_USERNAME="__token__" \
+    TWINE_PASSWORD="$PYPI_API_TOKEN" \
+    TWINE_REPOSITORY="$PYPI_REPO" \
+    twine upload "$STUBS_DIR/dist/*"
```
[the remark this answers is not in the document]
Comment (jmerle) [reply ci_build_stubs.sh]: Fixed in 45a8ef561c62c2053288d99e2fee72bbdc3e9a3c

**Review by gsalaz98 — APPROVED** [review]
@jaredbroad we'll need a few env var updates for these before merging:

* Add `PYPI_API_TOKEN`
* Remove the AWS keys from Travis.

Thank you @jmerle !

**Comment by jaredbroad** [comment]
Thank you @jmerle -- PYPI_API_TOKEN token set in Travis

**Pull request merged**